enterprise/app/workflows/github_import.tsx

```diff
@@ -0,0 +1,150 @@
+    });
+    this.setState({ importRequest });
+    rpcService.service
+      .createWorkflow(importRequest)
+      .then(() => {
+        // TODO: Show "Import succeeded" banner.
+        router.navigateToWorkflows();
+      })
+      .catch((error) => {
+        this.setState({ importRequest: null });
+        errorService.handleError(error);
+      });
+  }
+
+  private onClickShowMore() {
+    this.setState({ repoListLimit: this.state.repoListLimit + REPO_LIST_SHOW_MORE_INCREMENT });
+  }
+
+  private onClickWorkflowBreadcrumb(e: React.MouseEvent) {
+    e.preventDefault();
+    router.navigateToWorkflows();
+  }
+
+  private onClickImportOther(e: React.MouseEvent) {
+    e.preventDefault();
+    router.navigateTo("/workflows/new/custom");
+  }
+
+  private getImportedRepoUrls(): Set<string> {
+    return new Set(this.state.workflowsResponse.workflow.map((workflow) => workflow.repoUrl));
+  }
+
+  render() {
+    if (this.state.error) {
+      return <div className="error">{this.state.error}</div>;
+    }
+    if (!this.state.reposResponse || !this.state.workflowsResponse) {
+      return <div className="loading"></div>;
+    }
+    const alreadyImportedUrls = this.getImportedRepoUrls();
+    const isImporting = Boolean(this.state.importRequest);
+    return (
+      <div className="workflows-github-import">
+        <div className="shelf">
+          <div className="container">
+            <div className="breadcrumbs">
+              <span>
+                <a href="/workflows/" onClick={this.onClickWorkflowBreadcrumb.bind(this)}>
+                  Workflows
+                </a>
+              </span>
+              <span>Import GitHub repo</span>
+            </div>
+            <div className="title">Import GitHub repo</div>
+          </div>
+        </div>
+        <div className="container content-container">
+          <div className="repo-list">
+            {this.state.reposResponse.repo.slice(0, this.state.repoListLimit).map((repo) => {
+              const [owner, repoName] = parseOwnerRepo(repo.url);
+              return (
+                <div className="repo-item">
+                  <div className="owner-repo">
+                    <span>{owner}</span>
+                    <span className="repo-owner-separator">/</span>
+                    <span className="repo-name">{repoName}</span>
+                  </div>
+                  {alreadyImportedUrls.has(repo.url) ? (
+                    <div className="imported-indicator">
+                      <img src="/image/check.svg" title="Already imported" alt="" />
+                    </div>
+                  ) : isImporting && this.state.importRequest?.gitRepo?.repoUrl === repo.url ? (
+                    <div className="loading import-loading" />
+                  ) : (
+                    <FilledButton disabled={isImporting} onClick={this.onClickImport.bind(this, repo.url)}>
+                      Import
```

Comment:
maybe we can call this action "Set Up" or something instead of import? we're really just installing a webhook